test/test_singleton.cpp

```diff
@@ -1,74 +1,50 @@
 /////////1/////////2/////////3/////////4/////////5/////////6/////////7/////////8
-// test_singleton.cpp
-
-// (C) Copyright 2018 Robert Ramey - http://www.rrsd.com . 
+// test_singleton.cpp: test implementation of the singleton template
+//
+// - get_[const_]_instance returns the same instance everytime it is called
+// - is_destroyed returns false when singleton is active or uninitialized
+// - is_destroyed returns true when singleton is destructed
+// - the singleton is eventually destructed (no memory leak)
+
+// (C) Copyright 2002 Robert Ramey - http://www.rrsd.com .
 // Use, modification and distribution is subject to the Boost Software
 // License, Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
 // http://www.boost.org/LICENSE_1_0.txt)
-
-// should pass compilation and execution
-
-#include <iostream>
-#include <boost/serialization/singleton.hpp>
+// <gennadiy.rozental@tfn.com>
```

Comment:
Why is this line here?

Reply:
This is from #110 where I reverted changes made to a test I also changed. As explained in #110 the changed test currently in develop does rather test the compiler than the library which is why I reverted/overwrote it.

If this is objectionable I'll add my own version of `test_singleton.cpp` as `test_singleton2.cpp` or below into this file but that would mean redundant code.

Reply:
What I don't understand is what's Gennadiy Rozental's e-mail doing here, on a line of its own, it makes no sense to me.

Re test_singleton2.cpp, yes, I was going to suggest the same thing. I'd also drop the Travis changes, or just keep the essential changes to a minimum.

I also wonder whether the test will be cleaner if it's split into two, one testing a plain singleton, the other - inherited. There will be duplication, sure, but the array of states and the base class will vanish, so it could be more readable that way.

Reply:
As mentioned I reverted the original commit: https://github.com/boostorg/serialization/commit/8ca532a4d5b43f9e468a1a2a36dfb489e9ecc422#diff-b2f7572da5c9081e523924931012ab08

> Re test_singleton2.cpp, yes, I was going to suggest the same thing. I'd also drop the Travis changes, or just keep the essential changes to a minimum.

Some are required (`-fPIC`), I can drop the rest, as they are in #105 

> I also wonder whether the test will be cleaner if it's split into two, one testing a plain singleton, the other - inherited. There will be duplication, sure, but the array of states and the base class will vanish, so it could be more readable that way.

This would mean 2 separate compilation units. I wanted to have the 2 use cases (plain, inherited) next to each other, to show the similarity and difference. But I can split this in 2 files, you think, this is more readable.

Reply:
What I didn't like about this patch is that it's not clear which changes are necessary to make the test pass, and what aren't. The singleton_module changes, for instance, and a few other things.

But I now see that you're simply undoing https://github.com/boostorg/serialization/commit/e1893dd28c408bb25352596e2f91892b3d570164#diff-3f1e861be15fd4926f5e5b00388f2309.

Reply:
Yes got messy due to the failed attempt to fix this which is why I added an explicit revert commit. My Patch is a single commit here: https://github.com/boostorg/serialization/pull/105/commits/e0691b0af85b899a0bc9d9c4a94d763b2a25b003 (see the name)

The changes there are much less and mostly comments. Also compare the pre-1.65 version which that patch is based on.

Reply:
As requested I split the `is_destroyed` test into 2: https://github.com/boostorg/serialization/pull/129